src/Core/Managed/Shared/Extensibility/Implementation/BroadcastProcessor.cs

```diff
@@ -0,0 +1,148 @@
+﻿namespace Microsoft.ApplicationInsights.Shared.Extensibility.Implementation
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Microsoft.ApplicationInsights.Channel;
+    using Microsoft.ApplicationInsights.Extensibility;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing;
+
+    /// <summary>
+    /// An <see cref="ITelemetryProcessor"/> that asynchronously sends the data to multiple child telemetry processors.
+    /// </summary>
+    internal class BroadcastProcessor : ITelemetryProcessor, IDisposable
```

Comment:
I see your point, but this complexity buys you nice isolation between sinks/channels--see TelemetrySink tests FailingChannelDoesNotAffectOtherChannels and SlowChannelDoesNotAffectOtherChannels. Also it implicitly makes all sinks work in parallel, which we probably want regardless whether we call them synchronously or not. I felt these advantages outweigh the concerns about additional complexity.